crates/generate/src/types/int.rs

```diff
@@ -72,12 +72,23 @@ pub(super) fn define_int(names: &Names, name: &witx::Id, i: &witx::IntDatatype)
             }
 
             fn read(location: &wiggle_runtime::GuestPtr<'a, #ident>) -> Result<#ident, wiggle_runtime::GuestError> {
-                Ok(#ident(#repr::read(&location.cast())?))
+                let host_ptr =
+                    location.mem()
+                        .validate_size_align(location.offset(), Self::guest_align(), Self::guest_size())?;
+                Ok(unsafe { (host_ptr as *mut #ident).read() })
```

Comment:
I'm curious how come this changed? (the addition of `unsafe` here where it wasn't needed before)

Reply:
ah, i failed to revert my previous changes to the simplified version you found. will fix.